### PR TITLE
AXON-1980 UI for the invoke_subagents tool

### PR DESCRIPTION
**Pull request by Migelangel04**
### What Is This Change?

Added UI support for invoke_subagents in RovoDev chat.
https://www.loom.com/share/55cf2cf69f6f492c87c40c4740db47af

The new UI at a glance:
Screenshot 2026-03-19 at 12 51 44 PM

Screenshot 2026-03-19 at 12 51 58 PM

### How Has This Been Tested?

NOTE: For the invoke_subagents tool has inconsistent behavior due to an external factor that is being investigated current (nothing to do with these changes it was present before). The loom shows a bit a what I mean.

Aside from manual testing like shown in the loom, the two unit tests changed in this PR reflect the UI changes for the ToolCallItem component to render the correct dialog when the agent invokes subagents and displays it accordingly. The second file that was the util to test that the parsing of the data is correct and sends it correctly while testing the empty and missing fields edge cases (like missing description, subagent name, task name).

Basic checks:

- [X] `npm run lint`
- [X] `npm run test`

Advanced checks: 
- [ ] If Atlassian employee & Bitbucket changes: did you test with DC in mind? [See Instructions](https://www.loom.com/share/71e5d17734a547f68fd6128be6cd760e?sid=835e58a7-1240-498d-b2d7-fa7fdf8ffa36)

Recommendations:
- [X] Update the CHANGELOG if making a user facing change

---
Rovo Dev code review: <strong>Rovo Dev couldn't review this pull request</strong>
Upgrade to Rovo Dev Standard to continue using code review.

### CODE REVIEW

File: src/rovo-dev/ui/utils.tsx [hunk 1]
```diff
@@ -296,6 +302,18 @@ export function parseToolReturnMessage(
...
                 });
                 break;
+            case 'invoke_subagents':
+                const subagentArgs = safeJsonParse<SubagentArgs>(msg.toolCallMessage.args);
+                const subagentNames = subagentArgs?.subagent_names || [];
+                const taskNames = subagentArgs?.task_names || [];
+                for (let i = 0; i < subagentNames.length; i++) {
+                    resp.push({
+                        content: `Subagent: ${subagentNames[i]}`,
+                        title: taskNames[i] || '',
```
Comment (marcomura) [inline src/rovo-dev/ui/utils.tsx:322]: how would it look like with an empty title?

File: src/rovo-dev/ui/rovoDevView.tsx [hunk 2]
```diff
@@ -264,8 +266,25 @@ const RovoDevView: React.FC = () => {
                     const last = messages.at(-1);
                     if (last?.event_kind === 'tool-call') {
                         setPendingToolCallMessage(parseToolCallMessage(last.tool_name));
+                        if (last.tool_name === 'invoke_subagents') {
+                            try {
+                                const args = JSON.parse(last.args);
```
Comment (marcomura) [inline src/rovo-dev/ui/rovoDevView.tsx]: can we use `safeJsonParse` here instead of try/catch ?

File: src/rovo-dev/ui/rovoDevView.tsx [hunk 3]
```diff
@@ -264,8 +266,25 @@ const RovoDevView: React.FC = () => {
...
+                                const subagentNames: string[] = args.subagent_names || [];
+                                const taskNames: string[] = args.task_names || [];
+                                const tasks: SubagentInfo[] = subagentNames.map((name, i) => ({
+                                    subagentName: name,
+                                    taskName: taskNames[i] || '',
```
Comment (marcomura) [inline src/rovo-dev/ui/rovoDevView.tsx]: do we want to push a SubagentInfo object with taskName === '', or should we skip it if taskName is not assigned?

File: src/rovo-dev/ui/utils.tsx (hunk 1 above)
Comment (Migelangel04) [reply src/rovo-dev/ui/utils.tsx:322]: UI would look like:
Delegating tasks to subagents
Subagent: Subagent_Name ()
...

Basically from the picture above it would not have the title.

File: src/rovo-dev/ui/rovoDevView.tsx (hunk 2 above)
Comment (Migelangel04) [reply src/rovo-dev/ui/rovoDevView.tsx]: I will give it a go

File: src/rovo-dev/ui/rovoDevView.tsx (hunk 3 above)
Comment (Migelangel04) [reply src/rovo-dev/ui/rovoDevView.tsx]: I would say the taskName is the most important part of the object. The subagent name is not very descriptive as it is something generic like "Explore", "Search", etc and if it running multiple explores like in the PR description example, it does not give much info to the user. A better solution is to ensure each object has a taskName and do the logic as usual. If one is missing, send in an empty SubagentInfo array so that way all the user sees is "Delegating tasks to subagents"

**Review by marcomura — APPROVED** [review]
